File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 7/27/2026, 3:02:31 PM
class Solution {
public:
    int maxProfit(vector<int>& prices) 
    {
        int minprice=INT_MAX;
        int maxprofit=0;
        int profit;
        for(int i=0;i<prices.size();i++){
            if(prices[i]<minprice)
                minprice=prices[i];

            profit=prices[i]-minprice;
            if(profit>maxprofit)
                maxprofit=profit;
        }
        return maxprofit;
    }
};

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.java
// Last updated: 7/27/2026, 3:02:24 PM
class Solution {
    public int maxProfit(int[] prices) {
        int buy=prices[0];
        int profit=0; 

        for(int i=0;i<prices.length;i++){
            
             if(prices[i]<buy){
                buy=prices[i];
             }
             else if(prices[i]-buy>profit){
                profit=prices[i]-buy;
             }
        }
        return profit;
    }
}